### ISSUE TITLE
Fail to run some particular titles: Unknown server response

### ISSUE THREAD

**Issue by Soulou** [jpochyla/psst#92]
OS: Linux
Version: Current master at the time of the issue creation
Build: Release version

Which song:

Slowly - Supermodified - Amon Tobin

Logs from the terminal:

```
[2021-08-17T09:07:16Z ERROR psst_core::audio_key] audio key error
[2021-08-17T09:07:16Z ERROR psst_core::audio_player] failed to preload audio file, error while opening: Unknown server response
[2021-08-17T09:07:16Z ERROR psst_core::audio_key] audio key error
[2021-08-17T09:07:16Z ERROR psst_core::audio_player] failed to preload audio file, error while opening: Unknown server response
[2021-08-17T09:07:16Z ERROR psst_core::audio_key] audio key error
[2021-08-17T09:07:16Z ERROR psst_core::audio_player] failed to preload audio file, error while opening: Unknown server response
[2021-08-17T09:07:16Z ERROR psst_core::audio_key] audio key error
[2021-08-17T09:07:16Z ERROR psst_core::audio_player] failed to preload audio file, error while opening: Unknown server response
[...]
```

psst-gui just skips the song and start playing the next.

If you need more information, please just ask.

**Comment by jpochyla** [comment]
I've seen this as well, it's possible we're doing the track re-linking wrong.

**Comment by felixkiessling** [comment]
Same here. Got the message "error while loading: Permission denied (os error 13)" on macos.

**Comment by kunzjacq** [comment]
Indeed, having the same problem, I found that it is related to an incorrect logic in 

     fn is_restricted_in_region(restriction: &Restriction, country: &str)

in `metadata.rs`.
In its current form, this function returns false (i.e. the track is authorized to play) when there is a nonempty list of allowed countries and the user country code is not in that list, which is wrong. 

If this function is replaced by:

    fn is_restricted_in_region(restriction: &Restriction, country: &str) -> bool {
        if let Some(allowed) = &restriction.countries_allowed {
            return !(is_country_in_list(allowed.as_bytes(), country.as_bytes()));
        }
        if let Some(forbidden) = &restriction.countries_forbidden {
            return is_country_in_list(forbidden.as_bytes(), country.as_bytes()); 
        }
        false // should never be reached
    }


the errors dissapear.

**Comment by jpochyla** [comment]
Thanks a lot @kunzjacq! ❤️ Included in 105f6cc
